addressing issues 71, 68 and 69

ereshetova opened the pull request.

mmccool commented:
From the April 30 meeting:
- Thing Directory and Native Device threats being out of scope: necessary, since we don't have a specification we can refer to, but we should include some general comments and recommendations on how to mitigate these risks, referring to external references where appropriate.
- We should also be clear about what we mean by "out of scope": it doesn't mean they aren't important; it's specifically with respect to the Threat Model

ereshetova commented:
I have added a new column to the out-of-scope attack surfaces table, where I clarified why a certain attack surface is formally out of scope. I have also added there respective references to the new subsections in section 5, Recommended best practices, (empty for now), where we can collect and reference relevant high-level recommendations. I also fixed the "&" cases.
